# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/serviceworkers/test/sw_inter_sw_postmessage.js
# Commit: fa2538a4e354
# Full Hash: fa2538a4e354c1d9841b2a611def5843c1e383a6
# Author: Andrew Sutherland <asutherland@asutherland.org>
# Date: 2024-11-05 16:14:41
# Description:
#   Bug 1927247 - Regenerate Client Id for ServiceWorkers on termination. r=dom-worker-reviewers,webidl,smaug
#   
#   Bug 1544232 changed it so ServiceWorker globals used a ClientInfo and
#   Client Id created by the ServiceWorkerPrivate rather than creating a
#   random client id.  This allows the ServiceWorkerManager to reliably map
# ==============================================================================

diff -r 18af0608f7f1 -r fa2538a4e354 dom/serviceworkers/test/sw_inter_sw_postmessage.js
--- a/dom/serviceworkers/test/sw_inter_sw_postmessage.js	Tue Nov 05 05:18:37 2024 +0000
+++ b/dom/serviceworkers/test/sw_inter_sw_postmessage.js	Tue Nov 05 06:26:05 2024 +0000
@@ -156,6 +156,25 @@
         scope,
       });
       bc.postMessage(`${myId}:registered:${installId}`);
+    } else if (cmd === "workerref-hang") {
+      const topic = pieces[2];
+      globalThis.WorkerTestUtils.holdStrongWorkerRefUntilMainThreadObserverNotified(
+        topic
+      );
+      bc.postMessage(`${myId}:workerref-hung:${topic}`);
+    } else if (cmd === "block") {
+      const topic = pieces[2];
+      globalThis.WorkerTestUtils.blockUntilMainThreadObserverNotified(
+        topic,
+        // This callback is invoked once the observer has been registered.
+        () => {
+          bc.postMessage(`${myId}:blocking:${topic}`);
+        }
+      );
+    } else if (cmd === "notify-observer") {
+      const topic = pieces[2];
+      globalThis.WorkerTestUtils.notifyObserverOnMainThread(topic);
+      bc.postMessage(`${myId}:notified-observer:${topic}`);
     }
   } catch (ex) {
     console.error(ex);